Failing tests on macos

running `make test` on macos has some test failures - all of which seem to be related with `/tmp` path assertions

An example: 

```
____________________________________________________________________________________________________________________ test_validate_cachito_config_success ____________________________________________________________________________________________________________________

mock_isdir = <MagicMock name='isdir' id='4541205968'>, app = <Flask 'cachito.web.app'>

    @patch('os.path.isdir', return_value=True)
    def test_validate_cachito_config_success(mock_isdir, app):
        validate_cachito_config(app.config)
>       mock_isdir.assert_any_call('/tmp/cachito-archives/bundles')

tests/test_cachito_config.py:13:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <MagicMock name='isdir' id='4541205968'>, args = ('/tmp/cachito-archives/bundles',), kwargs = {}, expected = (('/tmp/cachito-archives/bundles',), {}), actual = [call('/var/folders/0h/28qktgvj01j9hp61hck2p7sc0000gn/T/cachito-archives/bundles')], cause = None
expected_string = "isdir('/tmp/cachito-archives/bundles')"

    def assert_any_call(self, *args, **kwargs):
        """assert the mock has been called with the specified arguments.

        The assert passes if the mock has *ever* been called, unlike
        `assert_called_with` and `assert_called_once_with` that only pass if
        the call is the most recent one."""
        expected = self._call_matcher((args, kwargs))
        actual = [self._call_matcher(c) for c in self.call_args_list]
        if expected not in actual:
            cause = expected if isinstance(expected, Exception) else None
            expected_string = self._format_mock_call_signature(args, kwargs)
            raise AssertionError(
                '%s call not found' % expected_string
>           ) from cause
E           AssertionError: isdir('/tmp/cachito-archives/bundles') call not found

/usr/local/opt/python/Frameworks/Python.framework/Versions/3.7/lib/python3.7/unittest/mock.py:942: AssertionError
```

--- a/tests/test_api_v1.py
+++ b/tests/test_api_v1.py
@@ -1,4 +1,6 @@
 # SPDX-License-Identifier: GPL-3.0-or-later
+import os
+import tempfile
 from unittest import mock
 
 import kombu.exceptions
@@ -429,7 +431,7 @@ def test_create_request_connection_error(mock_chain, app, auth_env, client, db):
 @mock.patch('flask.send_file')
 @mock.patch('cachito.web.api_v1.Request')
 def test_download_archive(mock_request, mock_send_file, mock_exists, client, app):
-    bundle_archive_path = '/tmp/cachito-archives/bundles/1.tar.gz'
+    bundle_archive_path = os.path.join(tempfile.gettempdir(), 'cachito-archives/bundles/1.tar.gz')
     mock_request.query.get_or_404().state.state_name = 'complete'
     mock_request.query.get_or_404().bundle_archive = bundle_archive_path
     mock_exists.return_value = True
@@ -493,8 +495,9 @@ def test_set_state(mock_rmtree, mock_exists, state, app, client, db, worker_auth
     assert fetched_request['state_history'][0]['state_reason'] == state_reason
     assert fetched_request['state_history'][0]['updated']
     assert fetched_request['state_history'][1]['state'] == 'in_progress'
-    mock_exists.assert_called_once_with('/tmp/cachito-archives/bundles/temp/1')
-    mock_rmtree.assert_called_once_with('/tmp/cachito-archives/bundles/temp/1')
+    temp_dir = os.path.join(tempfile.gettempdir(), 'cachito-archives/bundles/temp/1')
+    mock_exists.assert_called_once_with(temp_dir)
+    mock_rmtree.assert_called_once_with(temp_dir)
 
 
 def test_set_pkg_managers(app, client, db, worker_auth_env):
@@ -546,7 +549,12 @@ def test_set_state_stale(mock_remove, mock_exists, bundle_exists, app, client, d
     assert fetched_request['state'] == state
     assert fetched_request['state_reason'] == state_reason
     if bundle_exists:
-        mock_remove.assert_called_once_with('/tmp/cachito-archives/bundles/1.tar.gz')
+        mock_remove.assert_called_once_with(
+            os.path.join(
+                tempfile.gettempdir(),
+                'cachito-archives/bundles/1.tar.gz',
+            )
+        )
     else:
         mock_remove.assert_not_called()
 

--- a/tests/test_cachito_config.py
+++ b/tests/test_cachito_config.py
@@ -1,4 +1,6 @@
 # SPDX-License-Identifier: GPL-3.0-or-later
+import os
+import tempfile
 from unittest.mock import patch
 
 import pytest
@@ -10,7 +12,12 @@ from cachito.errors import ConfigError
 @patch('os.path.isdir', return_value=True)
 def test_validate_cachito_config_success(mock_isdir, app):
     validate_cachito_config(app.config)
-    mock_isdir.assert_any_call('/tmp/cachito-archives/bundles')
+    mock_isdir.assert_any_call(
+        os.path.join(
+            tempfile.gettempdir(),
+            'cachito-archives/bundles',
+        )
+    )
 
 
 @patch('os.path.isdir', return_value=True)

--- a/tests/test_workers/test_tasks.py
+++ b/tests/test_workers/test_tasks.py
@@ -2,6 +2,7 @@
 import os
 import pathlib
 import tarfile
+import tempfile
 from unittest import mock
 
 import pytest
@@ -30,7 +31,7 @@ def test_fetch_app_source(
     mock_set_request_state.assert_called_once_with(
         1, 'in_progress', 'Fetching the application source')
 
-    bundle_dir = '/tmp/cachito-archives/bundles/temp/1'
+    bundle_dir = os.path.join(tempfile.gettempdir(), 'cachito-archives/bundles/temp/1')
     mock_exists.assert_called_once_with(bundle_dir)
     if dir_exists:
         mock_makedirs.assert_not_called()
@@ -70,6 +71,8 @@ def test_fetch_gomod_source(
     mock_path_exists.return_value = contains_go_mod
     mock_resolve_gomod.return_value = sample_package, sample_deps_replace
     tasks.fetch_gomod_source(1, auto_detect, dep_replacements)
+    temp_app_dir = os.path.join(tempfile.gettempdir(), 'cachito-archives/bundles/temp/1/app')
+
     if expect_state_update:
         mock_set_request_state.assert_called_once_with(
             1, 'in_progress', 'Fetching the golang dependencies')
@@ -77,16 +80,24 @@ def test_fetch_gomod_source(
             1, sample_deps_replace, sample_env_vars, 'gomod', [sample_package])
 
     if auto_detect:
-        mock_path_exists.assert_called_once_with('/tmp/cachito-archives/bundles/temp/1/app/go.mod')
+        mock_path_exists.assert_called_once_with(
+            os.path.join(
+                tempfile.gettempdir(), 'cachito-archives/bundles/temp/1/app/go.mod'
+            )
+        )
         if contains_go_mod:
             mock_resolve_gomod.assert_called_once_with(
-                '/tmp/cachito-archives/bundles/temp/1/app', mock_request, dep_replacements,
+                temp_app_dir,
+                mock_request,
+                dep_replacements,
             )
         else:
             mock_resolve_gomod.assert_not_called()
     else:
         mock_resolve_gomod.assert_called_once_with(
-            '/tmp/cachito-archives/bundles/temp/1/app', mock_request, dep_replacements,
+            temp_app_dir,
+            mock_request,
+            dep_replacements,
         )
         mock_path_exists.assert_not_called()
 

